pkgs/applications/graphics/apitrace/default.nix

```diff
@@ -15,7 +15,42 @@ stdenv.mkDerivation rec {
   # of games -- so it's fine to use e.g. bundled snappy.
   buildInputs = [ libX11 procps python2 libdwarf qtbase qtwebkit ];
 
-  nativeBuildInputs = [ cmake ];
+  nativeBuildInputs = [ cmake wrapQtAppsHook ];
+
+  # Don't automatically wrap all binaries, I prefer to explicitly only wrap
+  # `qapitrace`.
+  dontWrapQtApps = true;
+
+  postFixup = ''
+    # Since https://github.com/NixOS/nixpkgs/pull/60985 OpenGL programs on NixOS are
+    # finding the opengl drivers in /run/opengl-driver thank to patched
+    # runpath, using `addOpenGLRunpath`. If they use a library loader such as
+    # `libglvnd`, this logic is already included in the library loader.
+    #
+    # Apitrace wraps programs using LD_PRELOAD wrappers (in $out/lib/apitrace/wrappers).
+    # Theses wrappers replace the library loader used by the original program
+    # by overriding `dlopen`.
+    #
+    # In order to find the correct OpenGL library, apitrace wrappers should
+    # have the correct rpath. This is done in the following lines.
+    # To stay simple, we add paths for 32 and 64 bits unconditionally. This
+    # have no impact on the final closure size.
```

Comment:
```suggestion

    # Since https://github.com/NixOS/nixpkgs/pull/60985, we add `/run-opengl-driver[-32]`
    # to the `RUNPATH` of dispatcher libraries `dlopen()` ing OpenGL drivers.
    # `RUNPATH` doesn't propagate throughout the whole application, but only from the module
    # performing the `dlopen()`.
    # 
    # Apitrace wraps programs by running them with `LD_PRELOAD` pointing to `.so` files
    # in $out/lib/apitrace/wrappers.
    # 
    # Theses wrappers effectively wrap the `dlopen()` calls from `libglvnd` and other dispatcher
    # libraries by themselves.
    # As `RUNPATH` doesn't propagate through the whole library, and they're now the
    # library doing the real `dlopen()`, they also need to have `/run-opengl-driver[-32]` added
    # to their `RUNPATH`.
    #
    # To stay simple, we add paths for 32 and 64 bits unconditionally.
    # This doesn't have an impact on closure size, and if the 32 bit drivers
    # are not available, that folder is ignored.
```